### ISSUE TITLE
Nested task include ignores conditional check

### ISSUE THREAD

**Issue by OrBaruk** [ansible/ansible#18206]
##### ISSUE TYPE
- Bug Report
##### COMPONENT NAME

Task Include
##### ANSIBLE VERSION

Affected versions:

```
ansible 2.1.2.0
  config file =
  configured module search path = Default w/o overrides
```

```
ansible 2.1.1.0
  config file =
  configured module search path = Default w/o overrides
```
##### OS / ENVIRONMENT

N/A
##### SUMMARY

When including a playbook that contains a taks includes using the nested task includes do no behave as expected regarding the conditional `when`.

The included tasks ignore the conditional checks and always execute, unless they are the first task in the included playbook.
##### STEPS TO REPRODUCE

Execute `ansible-playbook main.yml` with the following files in the root directory

```
# main.yml

---
- hosts: localhost
  tasks:
    - set_fact:
        my_var: False

- include: playbook.yml
  when: hostvars['localhost']['my_var']
```

```
# playbook.yml

---
- hosts: localhost
  tasks:
    - include: tasks_a.yml
    - include: tasks_b.yml
```

```
# tasks_a.yml

---
- debug: msg="Hello from A!"
```

```
# tasks_b.yml

---
- debug: msg="Hello from B!"
```
##### EXPECTED RESULTS

Both included tasks are skipped due to conditional check from the parent playbook. This behavior is observed with the version `ansible 2.1.0.0`

```
PLAYBOOK: main.yml *************************************************************
2 plays in main.yml

PLAY [localhost] ***************************************************************

TASK [setup] *******************************************************************
ok: [localhost]

TASK [set_fact] ****************************************************************
task path: /home/dev/ansbug/main.yml:5
ok: [localhost] => {"ansible_facts": {"my_var": false}, "changed": false, "invocation": {"module_args": {"my_var": false}, "module_name": "set_fact"}}

PLAY [localhost] ***************************************************************

TASK [setup] *******************************************************************
ok: [localhost]

TASK [include] *****************************************************************
task path: /home/dev/ansbug/playbook.yml:5
skipping: [localhost] => {"changed": false, "skip_reason": "Conditional check failed", "skipped": true}

TASK [include] *****************************************************************
task path: /home/dev/ansbug/playbook.yml:6
skipping: [localhost] => {"changed": false, "skip_reason": "Conditional check failed", "skipped": true}

PLAY RECAP *********************************************************************
localhost                  : ok=3    changed=0    unreachable=0    failed=0
```
##### ACTUAL RESULTS

Only the first included task obey the conditional check from the parent playbook.

```
PLAYBOOK: main.yml *************************************************************
2 plays in main.yml

PLAY [localhost] ***************************************************************

TASK [setup] *******************************************************************
ok: [localhost]

TASK [set_fact] ****************************************************************
task path: /home/dev/ansbug/main.yml:5
ok: [localhost] => {"ansible_facts": {"my_var": false}, "changed": false, "invocation": {"module_args": {"my_var": false}, "module_name": "set_fact"}}

PLAY [localhost] ***************************************************************

TASK [setup] *******************************************************************
ok: [localhost]

TASK [debug] *******************************************************************
task path: /home/dev/ansbug/tasks_a.yml:3
skipping: [localhost] => {"changed": false, "skip_reason": "Conditional check failed", "skipped": true}

TASK [debug] *******************************************************************
task path: /home/dev/ansbug/tasks_b.yml:3
ok: [localhost] => {
    "msg": "Hello from B!"
}

PLAY RECAP *********************************************************************
localhost                  : ok=4    changed=0    unreachable=0    failed=0
```

**Comment by OrBaruk** [comment]
Can confirm that problem is fixed.

Tested on the following version:

```
ansible-playbook 2.2.0.0 (detached HEAD 09fc911a48) last updated 2016/11/08 12:58:52 (GMT -200)
  lib/ansible/modules/core: (detached HEAD 164225aa43) last updated 2016/11/08 12:58:56 (GMT -200)
  lib/ansible/modules/extras: (detached HEAD 18bb736cc2) last updated 2016/11/08 12:58:56 (GMT -200)
  config file = /Users/Or/tmp/ansible.cfg
  configured module search path = Default w/o overrides
```

**Comment by jimi-c** [comment]
@OrBaruk I was not seeing this fixed yet. Using your example (on both devel and stable-2.2), I see the following:

<pre>
TASK [debug] *********************************************************************************************************************************************************************************
task path: /home/jamesc/testing/18206/tasks_a.yml:2
ok: [localhost] => {
    "msg": "Hello from A!"
}
TASK [debug] *********************************************************************************************************************************************************************************
task path: /home/jamesc/testing/18206/tasks_b.yml:2
skipping: [localhost] => {"changed": false, "skip_reason": "Conditional check failed", "skipped": true}
</pre>

when both should be skipped. Can you confirm this?

**Comment by OrBaruk** [comment]
Indeed it not fixed yet (both devel and stable-2.2).

I had changed some of my example files locally and it was not reproducing correctly the bug anymore. Sorry for the confusion.
